src/ethereum/berlin/vm/instructions/system.py

```diff
@@ -290,111 +300,72 @@ def call(evm: Evm) -> None:
     evm :
         The current EVM frame.
     """
-    from ...vm.interpreter import STACK_DEPTH_LIMIT, process_message
-
-    gas = pop(evm.stack)
+    # STACK
+    gas = Uint(pop(evm.stack))
     to = to_address(pop(evm.stack))
     value = pop(evm.stack)
-    memory_input_start_position = Uint(pop(evm.stack))
+    memory_input_start_position = pop(evm.stack)
     memory_input_size = pop(evm.stack)
-    memory_output_start_position = Uint(pop(evm.stack))
+    memory_output_start_position = pop(evm.stack)
     memory_output_size = pop(evm.stack)
-    ensure(not evm.message.is_static or value == U256(0), WriteInStaticContext)
 
-    gas_input_memory = calculate_gas_extend_memory(
-        evm.memory, memory_input_start_position, memory_input_size
-    )
-    evm.gas_left = subtract_gas(evm.gas_left, gas_input_memory)
-    extend_memory(evm.memory, memory_input_start_position, memory_input_size)
-    gas_output_memory = calculate_gas_extend_memory(
-        evm.memory, memory_output_start_position, memory_output_size
-    )
-    evm.gas_left = subtract_gas(evm.gas_left, gas_output_memory)
-    extend_memory(evm.memory, memory_output_start_position, memory_output_size)
-    call_data = memory_read_bytes(
-        evm.memory, memory_input_start_position, memory_input_size
-    )
+    # GAS
+    extend_memory(evm, memory_input_start_position, memory_input_size)
+    extend_memory(evm, memory_output_start_position, memory_output_size)
+    is_account_alive = account_exists(
+        evm.env.state, to
+    ) and not is_account_empty(evm.env.state, to)
 
     if to in evm.accessed_addresses:
         access_gas_cost = GAS_WARM_ACCESS
     else:
-        access_gas_cost = GAS_COLD_ACCOUNT_ACCESS
         evm.accessed_addresses.add(to)
+        access_gas_cost = GAS_COLD_ACCOUNT_ACCESS
 
-    is_account_alive = account_exists(
-        evm.env.state, to
-    ) and not is_account_empty(evm.env.state, to)
     create_gas_cost = (
-        U256(0) if is_account_alive or value == 0 else GAS_NEW_ACCOUNT
+        Uint(0) if is_account_alive or value == 0 else GAS_NEW_ACCOUNT
```

Comment:
I'm really not happy with how `black` is formatting `is_account_alive = ...` :rofl: 

Could we do something like:

```suggestion
        Uint(0) if value == 0 or account_alive(evm.env.state, to) else GAS_NEW_ACCOUNT
```

The `value == 0` check is pretty cheap to do, so we should do it first to short circuit when able.